ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfiguration.java

```diff
@@ -102,4 +113,29 @@ private JsonRpcResponse respondWith(
       final EngineExchangeTransitionConfigurationResult transitionConfiguration) {
     return new JsonRpcSuccessResponse(requestId, transitionConfiguration);
   }
+
+  // Handler is a member func in order to refer to our vertx instance,
+  // qosTimeout is a parameter for testing
+  Handler<Long> qosHandler(final long qosTimeout) {
+    return z -> {
+      if (getLastCallMillis() + qosTimeout < System.currentTimeMillis()) {
+        logQosFailure(qosTimeout);
+      }
+      resetQosHandler(qosTimeout, qosHandler(qosTimeout));
+    };
+  }
+
+  void resetQosHandler(final long qosTimeout, final Handler<Long> qosHandler) {
+    Optional.ofNullable(qosTimerId.get()).ifPresent(timerVertx::cancelTimer);
```

Comment:
`AtomicLong.get()` returns `long`, so it never can be `null` and `cancelTimer` will be called every time. We could init `qosTimerId` with `null` and check directly for that. Not sure if there are other solutions that would be cleaner.

Reply:
Good point - I started with AtomicReference, and changed it to AtomicLong.  I will init the AtomicLong with an explicit timerId like Long.MAX_VALUE and just cancel it